jobs/integration/test_opa.py

```diff
@@ -0,0 +1,439 @@
+                "juju",
+                "remove-storage-pool",
+                "-m",
+                tools.k8s_connection,
+                storage_pool_name,
+            )
+
+    @pytest.fixture(scope="class")
+    async def update_status_interval(self, tools, k8s_model):
+        interval = await tools.run(
+            "juju",
+            "model-config",
+            "-m",
+            tools.k8s_connection,
+            "update-status-hook-interval",
+        )
+        interval = interval[0].strip()
+        try:
+            log("Decrease update status hook interval")
+            await tools.run(
+                "juju",
+                "model-config",
+                "-m",
+                tools.k8s_connection,
+                "update-status-hook-interval=5s",
+            )
+            yield
+        finally:
+            await tools.run(
+                "juju",
+                "model-config",
+                "-m",
+                tools.k8s_connection,
+                f"update-status-hook-interval={interval}",
+            )
+
+    @pytest.fixture(scope="class")
+    async def opa_controller_manager(self, model, k8s_model, tools):
+        try:
+            log("Deploying the webhook charm")
+            webhook = await k8s_model.deploy(
+                "gatekeeper-controller-manager",
+                channel=tools.charm_channel,
+                trust=True,
+            )
+            yield webhook
+        finally:
+            log("Deleting the gatekeeper charm")
+            await tools.run(
+                "juju",
+                "remove-application",
+                "-m",
+                tools.k8s_connection,
+                "--force",
+                "gatekeeper-controller-manager",
+            )
+            await self.wait_for_app_removed("gatekeeper-controller-manager", k8s_model)
+
+    @pytest.fixture(scope="class")
+    async def opa_audit(self, model, k8s_model, tools, storage_pool):
+        try:
+            log("Deploying the audit charm")
+            audit = await k8s_model.deploy(
+                "gatekeeper-audit",
+                channel=tools.charm_channel,
+                trust=True,
+                storage={"audit-volume": {"pool": storage_pool}},
+                config={"audit-interval": 1},
+            )
+            yield audit
+        finally:
+            log("Deleting the audit charm")
+            await tools.run(
+                "juju",
+                "remove-application",
+                "-m",
+                tools.k8s_connection,
+                "--force",
+                "--destroy-storage",
+                "gatekeeper-audit",
+            )
+            await self.wait_for_app_removed("gatekeeper-audit", k8s_model)
+
+    async def _validate_audit_actions(self, unit):
+        log("Running list-violations action")
+        total_violations = 0
+        while total_violations == 0:
+            action = await unit.run_action("list-violations")
```

Comment:
there are changes coming between juju 2.9.11 and 3.0.1  that mean we should use `juju_run_action` rather than `unit.run_action`.  I'll save you the hassle but python-lib-juju is making a breaking change to their API so i'm trying to shove all the action running logic into there. 

There's plenty of examples of how to use `juju_run_action(...)` in validation.py